src/urlMatcherFactory.js

```diff
@@ -917,7 +917,8 @@ function $UrlMatcherFactory() {
     }
 
     function getType(config, urlType, location) {
-      if (config.type && urlType) throw new Error("Param '"+id+"' has two type configurations.");
+      if (config.type && urlType.name !== 'string') throw new Error("Param '"+id+"' has two type configurations.");
+      if (config.type && urlType.name === 'string' && $types[config.type]) return $types[config.type];
```

Comment:
Since `urlType` is optional, this code will break on url-less parameters.  I liked the direction of your previous logic better, where it checks if config.type is a string.

`angular.isString` is a handy utility for checking if something is a string or not